### PR TITLE
introduce flag for insecure communication

### LINKED ISSUE #13: It doesn't seem to be possible to configure WithInsecure for an opentelemetry collector endpoint

### PR DESCRIPTION
**Pull request by vroldanbet**
closes https://github.com/jzelinskie/cobrautil/issues/13

this is particularly useful in development environments so that folks don't have to setup TLS in their collector

### CODE REVIEW

File: otel.go [hunk 1]
```diff
@@ -38,6 +38,7 @@ func RegisterOpenTelemetryFlags(flags *pflag.FlagSet, flagPrefix, serviceName st
 	flags.String(prefixed("endpoint"), "", "OpenTelemetry collector endpoint - the endpoint can also be set by using enviroment variables")
 	flags.String(prefixed("service-name"), serviceName, "service name for trace data")
 	flags.String(prefixed("trace-propagator"), "w3c", `OpenTelemetry trace propagation format ("b3", "w3c", "ottrace"). Add multiple propagators separated by comma.`)
+	flags.String(prefixed("insecure"), "false", `OpenTelemetry communicate with the collector with HTTP instead of HTTPS.`)
```
Comment (jzelinskie) [inline otel.go]: Trying to think of a better flag description

"connect to the OpenTelemetry collector in plaintext"

**Comment by jzelinskie** [comment]
Let's double check there isn't a way to do this with environment variables first.

**Comment by vroldanbet** [comment]
Closing - you can achieve this by using OTELs `OTEL_EXPORTER_OTLP_ENDPOINT` env var like `http://localhost:4318`. Otel library calls `WithInsecure` if `http` is set.

https://github.com/open-telemetry/opentelemetry-go/blob/main/exporters/otlp/otlptrace/internal/otlpconfig/envconfig.go#L97-L104

**Comment by vroldanbet** [comment]
@jzelinskie I've reopened this. Even though it's possible to enable insecure communication via OTEL environment variables, it's not immediately obvious to users of applications using `cobrautil`. For example SpiceDB offers `--help` flag with all options, including those related to OTEL provided by `cobrautil`. It would be a better UX - we cannot expect folks to go and check the opentelemetry go library source code to figure out they can achieve it with `OTEL_EXPORTER_OTLP_ENDPOINT`.

Until it's decided to deprecate cobrautil/otel flags in favour of OTEL's native envs, we should provide a cohesive interface in cobrautil.

**Comment by jzelinskie** [comment]
I think that's reasonable, is there an equivalent for Jaeger though? Because the cobrautil flag is not tied to a particular provider, ideally it can also trigger the behavior for Jaeger.

**Comment by vroldanbet** [comment]
@jzelinskie 

>I think that's reasonable, is there an equivalent for Jaeger though? Because the cobrautil flag is not tied to a particular provider, ideally it can also trigger the behavior for Jaeger.

in Jaeger the endpoint must have the protocol as prefix, so it's defined through that. Nothing in the API showed the equivalent, but I can dig a bit more

EDIT: I couldn't wind any equivalent in the otel Jaeger exporter. It would seem to be delegated to the `http.Client`, which figures out based on the endpoint URL

File: otel.go [hunk 2]
```diff
@@ -83,7 +86,15 @@ func OpenTelemetryRunE(flagPrefix string, prerunLevel zerolog.Level) CobraRunFun
 			serviceName = stringz.Default(serviceName, MustGetString(cmd, "otel-jaeger-service-name"), "", cmd.Flags().Lookup(prefixed("service-name")).DefValue)
 
 			var opts []jaeger.CollectorEndpointOption
+
 			if endpoint != "" {
+				parsed, err := url.Parse(endpoint)
+				if err != nil {
+					return err
```
Comment (jzelinskie) [inline otel.go]: just need to wrap this error

Comment (vroldanbet) [reply otel.go]: done!

**Review by jzelinskie — APPROVED** [review]

**Pull request merged**